pkg/platform/runtime/buildscript/buildexpression.go

```diff
@@ -0,0 +1,151 @@
+package buildscript
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/rtutils/p"
+	buildexpression "github.com/ActiveState/cli/pkg/buildexpression/parser"
+	"github.com/ActiveState/cli/pkg/platform/api/graphql/model/buildplanner"
+)
+
+func FromBuildExpression(tree *buildexpression.Tree) *Script {
+	script := &Script{&Let{}, &In{}}
+	// tree.Root.Children() contains the following 7 nodes: { let : <binding> , in <in expr>
+
+	letBinding := tree.Root.Children()[1].Children()[3]
```

Comment:
Note that this and a lot of other index operations with magic numbers are due to how the parser produces the AST. It is subject to change until finalized.

Reply:
It might be nice to include a link to some doc and/or a comment to clarify the basis for the magic numbers. To make them non-magical, consider defining some constants where the names act as clarification.

Reply:
After discussion, we're tossing out the custom build expression parser and lexer in favor of unmarshaling the build expression JSON directly and manipulating it. This file has been completely rewritten.